datafusion/core/src/physical_plan/sorts/sort.rs

```diff
@@ -1097,7 +1097,7 @@ mod tests {
                 .with_fetch(fetch),
             );
 
-            let result = collect(sort_exec.clone(), task_ctx).await?;
+            let result = collect(sort_exec.clone(), task_ctx.clone()).await?;
```

Comment:
`task_ctx` is simply moved `task_ctx`. Before the stream future is resolved, it is dropped early.